src/indicators/singleValuedIndicator.js

```diff
@@ -0,0 +1,75 @@
+(function(d3, fc) {
+    'use strict';
+
+    fc.indicators.singleValuedIndicator = function(algorithm) {
+
+        var xScale = d3.time.scale(),
+            yScale = d3.scale.linear(),
+            yValue = function(d) { return d.close; },
+            xValue = function(d) { return d.date; };
+
+        var outputLine = fc.series.line();
+
+        var singleValuedIndicator = function(selection) {
+
+            algorithm.value(yValue);
+
+            outputLine.xScale(xScale)
+                .yScale(yScale)
+                .xValue(xValue)
+                .yValue(function(d, i) { return d.singleValuedIndicator; });
+
+            selection.each(function(data) {
+
+                data = d3.zip(data, algorithm(data))
```

Comment:
I know I came up with this pattern but I'm now thinking it should be changed to make it's intent clearer. Instead of a map (which implies that nothing is going to be modified), I wonder if it should be a forEach -
```js
 d3.zip(data, algorithm(data))
  .forEach(function(tuple) {
    tuple[0].singleValuedIndicator = tuple[1];
  });
```